tests/foreman/cli/test_repository.py

```diff
@@ -75,68 +83,332 @@
+        for url in (FAKE_1_YUM_REPO, FAKE_3_YUM_REPO, FAKE_4_YUM_REPO)
+    ],
+    'yum_sync_1': [{'content-type': 'yum', 'url': FAKE_1_YUM_REPO}],
+    'yum_url': [{'content-type': 'yum', 'url': url} for url in YUM_REPOS],
+}
+REPO_PARAMS = {k: parametrized(v) for k, v in REPO_OPTIONS.items()}
 
-    org = None
-    product = None
 
-    def setUp(self):
-        """Tests for Repository via Hammer CLI"""
+REPO_CRED_OPTIONS = {
+    'yum_http_invalid': [
+        ({'content-type': 'yum', 'url': FAKE_5_YUM_REPO.format(cred['login'], cred['pass'])}, cred)
+        for cred in VALID_CREDS
+        if not cred['http_valid']
+    ],
+}
+REPO_CRED_PARAMS = {k: parametrized(v) for k, v in REPO_CRED_OPTIONS.items()}
 
-        super().setUp()
 
-        if RepositoryTestCase.org is None:
-            RepositoryTestCase.org = make_org(cached=True)
-        if RepositoryTestCase.product is None:
-            RepositoryTestCase.product = make_product_wait(
-                {'organization-id': RepositoryTestCase.org['id']}
-            )
+def _get_image_tags_count(repo):
+    return Repository.info({'id': repo['id']})
 
-    def _make_repository(self, options=None):
-        """Makes a new repository and asserts its success"""
-        if options is None:
-            options = {}
 
-        if options.get('product-id') is None:
-            options['product-id'] = self.product['id']
+def _validated_image_tags_count(repo):
+    """Wrapper around Repository.info(), that returns once
+    container-image-tags in repo is greater than 0.
+    Needed due to BZ#1664631 (container-image-tags is not populated
+    immediately after synchronization), which was CLOSED WONTFIX
+    """
+    wait_for(
+        lambda: int(_get_image_tags_count(repo=repo)['content-counts']['container-image-tags'])
+        > 0,
+        timeout=30,
+        delay=2,
+        logger=logging.getLogger('robottelo'),
+    )
+    return _get_image_tags_count(repo=repo)
 
-        return make_repository(options)
 
-    def _get_image_tags_count(self, repo=None):
-        repo_detail = Repository.info({'id': repo['id']})
-        return repo_detail
+def _org(request):
+    org = make_org()
 
-    def _validated_image_tags_count(self, repo=None):
-        """Wrapper around Repository.info(), that returns once
-        container-image-tags in repo is greater than 0.
-        Needed due to BZ#1664631 (container-image-tags is not populated
-        immediately after synchronization), which was CLOSED WONTFIX
-        """
-        wait_for(
-            lambda: int(
-                self._get_image_tags_count(repo=repo)['content-counts']['container-image-tags']
-            )
-            > 0,
-            timeout=30,
-            delay=2,
-            logger=self.logger,
-        )
-        return self._get_image_tags_count(repo=repo)
+    @request.addfinalizer
+    def _cleanup():
+        if Org.exists(search=('id', org['id'])):
+            org_cleanup(org['id'])
```

Comment:
@rplevka has raised concerns about deleting organizations, because creating entities as admin user puts the entity in the scope of every organization.  Deleting the organization could have impact on running tests, especially in the current CI structure where xdist is used against a single satellite target. We will likely have to leave this off for now, and fix the behavior at a fundamental level before starting to clean up organizations during test execution.

Reply:
just for the record - this is not a case of every entity. IIRC, this was happening with e.g. capsules.

Reply:
Removed cleanup code pending further discussion.